test/test_package_manifest.rb

```diff
@@ -155,6 +156,14 @@ def parse_dependency(xml)
                     dependency = parse_dependency("<package><depend package='test' optional='1'/></package>")
                     assert dependency.optional
                 end
+                it "is not included if the condition is false" do
+                    manifest = PackageManifest.parse(pkg, "<package><depend package='test' condition='1==0'/></package>")
```

Comment:
You are only testing constants, not that you can evaluate variables that have been set in the chosen variable source (i.e. env or conf)

Reply:
More complex expressions are already tested in test_ros_condition_parser.rb.

Reply:
> More complex expressions are already tested in test_ros_condition_parser.rb.

But yeah, should also test if a variable actually manages to work its way from (in this case) manifest.package.ws.env into the condition parser.

Reply:
That was my point exactly ;-)